src/link.js

```diff
@@ -33,36 +42,20 @@ export async function loginWithLink({ token, uuid, redirect } = {}) {
       },
       {
         headers: getMfaHeaders(),
-        params: getPkceRequestQueryParams()
+        params: getPkceRequestQueryParams(),
       }
     );
 
-    if (data.hasOwnProperty("tokens")) {
-      clearMfa();
-      setCookiesAndTokens(data.tokens);
-      await exchange(data);
-      handleRedirect({ redirect, data });
-      return data;
-    }
-
-    if (data.hasOwnProperty("firstFactorToken")) {
-      handleMfaRequired(data);
-      return data;
-    }
-
-    if (data.authorizationCode) {
-      const url = redirect || data.redirectTo;
-      if (url) {
-        redirectWithPkce(url, data.authorizationCode);
-        return;
-      } else {
-        // We can't exchange the authorizationCode for tokens, because we don't have the verifier code
-        // that matches our challenge code.
-        throw new Error("Received a PKCE (mobile auth) response from the server, but no redirect was provided. Please set the redirect to the app that initiated the request.")
-      }
-    }
-
-    throw new Error("Problem logging in.");
+    // Handle the API response to the login request
+    return handleLoginResponse({
+      data,
+      redirect,
+      handleUpstreamResponse,
+      handleMfaRequired,
+      handlePkceRequired,
+      handleTokens,
+      handleRedirect,
+    });
```

Comment:
This method was used to standardize handling of the API response across various signup & login methods.